jinahub/indexers/SimpleIndexer/simple_indexer.py

```diff
@@ -93,10 +94,11 @@ def search(
         top_k = int(parameters.get('top_k', self.default_top_k))
         flat_docs.match(
             self._docs,
-            metric=lambda q_emb, d_emb, _: self.distance(
+            metric=lambda q_emb, d_emb: self.distance(
                 _ext_A(_norm(q_emb)), _ext_B(_norm(d_emb))
             ),
             limit=top_k,
+            metric_name=self.metric_name
```

Comment:
Is it possible that you can point me to where this `metric_name` is needed in examples?

Reply:
match is to be handled like in #79

Reply:
The problem we have here is that we pass a lambda function to the .match() method as a distance metric. Therefore, the name of the metric is not 'cosine' as expected but <lambda> instead. Thie means I have to retrieve the score like so: `doc.scores['<lambda>]'` which is not at all what we want.